storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -708,8 +708,17 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   }
 
   // check if xtrabackup_info, almost the last file exists in backup
-  if (std::count(object_list.begin(), object_list.end(),
-                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
+  std::string last_file_expecting = backup_name + "/xtrabackup_info";
+  bool file_found = false;
+  for (auto cur_file : object_list) {
+    if (cur_file.size() > last_file_expecting.size() &&
+        cur_file.substr(0, last_file_expecting.size())
+                .compare(last_file_expecting) == 0) {
+      file_found = true;
+      break;
+    }
+  }
+  if (!file_found) {
     msg_ts("%s: Upload failed: backup is incomplete.\n", my_progname);
     return false;
```

Comment:
Not clear what you are trying to do? before this change you were looking for the file "xtrabackup_info.00000000000000000000" from the cloud backup dir. What is the new strategy for compressed files? Do they end with .xbcrypt extension? Can you please explain with example?

Reply:
they end with xtrabackup_info.gz.00000000000 so now we are just checking if there exists file initial substring is xtrabackup_info. So now xtrabackup_info.gz.00000000000 or xtrabackup_info.00000000000000000000 both will pass the check

Reply:
thanks for explanation